model/billing_info.rb

```diff
@@ -11,24 +11,29 @@ class BillingInfo < Sequel::Model
 
   include ResourceMethods
 
+  def stripe_customer
+    return unless (Stripe.api_key = Config.stripe_secret_key)
+    @stripe_customer ||= Stripe::Customer.retrieve(stripe_id)
+  end
+
   def stripe_data
-    if (Stripe.api_key = Config.stripe_secret_key)
-      @stripe_data ||= begin
-        data = Stripe::Customer.retrieve(stripe_id)
-        address = data["address"]
-        {
-          "name" => data["name"],
-          "email" => data["email"],
-          "address" => [address["line1"], address["line2"]].compact.join(" "),
-          "country" => address["country"],
-          "city" => address["city"],
-          "state" => address["state"],
-          "postal_code" => address["postal_code"],
-          "tax_id" => data["metadata"]["tax_id"],
-          "company_name" => data["metadata"]["company_name"]
-        }
-      end
-    end
+    return unless (data = stripe_customer)
+    address = data["address"] || {}
+    {
+      "name" => data["name"],
+      "email" => data["email"],
+      "address" => [address["line1"], address["line2"]].compact.join(" "),
+      "country" => address["country"],
+      "city" => address["city"],
+      "state" => address["state"],
+      "postal_code" => address["postal_code"],
+      "tax_id" => data["metadata"]["tax_id"],
+      "company_name" => data["metadata"]["company_name"]
+    }
+  end
+
+  def has_address?
+    !stripe_customer&.[]("address").nil?
```

Comment:
Since we removed the cache clearing, I don't think we need the extra `stripe_customer` method.


```suggestion
    !stripe_data&.[]("address").empty?
```

Reply:
Done.